### DISCUSSION TITLE
How to run migration when using docker

### DISCUSSION THREAD

**Discussion by LadIQe** [mikro-orm/mikro-orm#1468] [Q&A]
Hello,

I moved from TypeORM to Mikro-ORM for brighter future, but now Im regretting it. I have docker config, where I have 2 services, one for db, one for server.

I already figured out how to properly set configs for initializations and for CLI. But, migrations are not working at all. When I type

`migration:create -i`

there is table for migrations, but when I type

`migration:create`

there is no actual table for it, even migration table has no data about it. I have migration file in my project, but thats all. Also, I can run several times "create" command without any issue.

Migration list is empty, but migration pending has record about my migration. How to solve it? For what purpose is pending list?

Thank you

**Comment by B4nan** [comment]
Sounds like you are confusing migrations with schema generator. `migration create` will create the migration file (containing the diff from the schema generator) - it will not execute anything. You need to call `migration up` for that (which will execute all pending migrations). 

Also note that you don't need initial migration usually - that flag is only when you already have a schema and want to start using migrations. It will create the initial migration (again, just a file), and mark it as executed (if you have a schema). 

https://mikro-orm.io/docs/migrations/
https://mikro-orm.io/docs/schema-generator
